contracts/vaults/Withdraw.sol

```diff
@@ -60,12 +57,18 @@ contract Withdraw is Swap {
         user.rewardDebt = 0;
         uint256 fee = factoredBalance * pool.withdrawalFee / HUNDRED_PERCENT;
         pool.balance -= factoredBalance;
-        if (fee > 0) {
-            pool.lpToken.safeTransfer(governance(), fee);
-        }
-        pool.lpToken.safeTransfer(msg.sender, factoredBalance - fee);
+        safeWithdrawPoolToken(pool.lpToken, factoredBalance, fee);
         emit EmergencyWithdraw(msg.sender, _pid, factoredBalance);
     }
+
+    function safeWithdrawPoolToken(IERC20 _lpToken, uint256 _totalAmount, uint256 _fee) internal {
+        uint256 balanceBefore = _lpToken.balanceOf(address(this));
+        if (_fee > 0) {
+            _lpToken.safeTransfer(governance(), _fee);
+        }
+        _lpToken.safeTransfer(msg.sender, _totalAmount - _fee);
+        require(balanceBefore - _lpToken.balanceOf(address(this)) == _totalAmount, "HVE45");
```

Comment:
So I do not think we need this require. I mean, I can think of 2 scenarios here: 
- either the require fails for a good reason, in which case our users cannot withdraw, and it is our fault
- the require fails for a bad reason, in which case the token is problematic, and it can fail in many other ways, perhaps much worse, and this require is not saving us from any problem
So my opinion we should drop this requirement